nemo/collections/llm/recipes/callbacks/default.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+from nemo_run import Config, cli
+from pytorch_lightning.callbacks.callback import Callback
+
+from nemo.utils.import_utils import safe_import
+
+res_module, HAVE_RES = safe_import('nvidia_resiliency_ext.ptl_resiliency')
```

Comment:
Since this will be in all recipes now, and we currently don't have 'nvidia_resiliency_ext' as a hard dependency, should we return 'None' or some dummy callback if the package isn't installed?

Reply:
That would suppress the ability to catch issues with 'nvidia_resiliency_ext'. It will also make debugging issues with it harder for developers. 

Some options come to mind - 
1. Make 'nvidia_resiliency_ext' a hard dependency for all training jobs
2. Disable resiliency features by default since it is not a hard requirement and add documentation on how to enable it, if needed.

Reply:
I think long-term, we want to do 1. but we need them to support installing on Windows for that. 2. is probably best until then.